spec/rubocop/cop/style/double_negation_spec.rb

```diff
@@ -48,6 +48,249 @@ def foo?
+            do_something
+          end
+        end
+      RUBY
+
+      expect_correction(<<~RUBY)
+        def foo?
+          unless condition_foo?
+            !foo.nil?
+            do_something
+          end
+        end
+      RUBY
+    end
+
+    it 'registers an offense and corrects for `!!` when not return location' \
+       'and using `if`, `elsif`, and `else`' do
+      expect_offense(<<~RUBY)
+        def foo?
+          if condition_foo?
+            !!foo
+            ^ Avoid the use of double negation (`!!`).
+            do_something
+          elsif condition_bar?
+            !!bar
+            ^ Avoid the use of double negation (`!!`).
+            do_something
+          else
+            !!baz
+            ^ Avoid the use of double negation (`!!`).
+            do_something
+          end
+        end
+      RUBY
+
+      expect_correction(<<~RUBY)
+        def foo?
+          if condition_foo?
+            !foo.nil?
+            do_something
+          elsif condition_bar?
+            !bar.nil?
+            do_something
+          else
+            !baz.nil?
+            do_something
+          end
+        end
+      RUBY
+    end
+
+    it 'registers an offense and corrects for `!!` with hash when not return location' \
+       'and using `if`, `elsif`, and `else`' do
+      expect_offense(<<~RUBY)
+        def foo?
+          if condition_foo?
+            { foo: !!foo0, bar: bar0, baz: baz0 }
+                   ^ Avoid the use of double negation (`!!`).
+            do_something
+          elsif condition_bar?
+            { foo: !!foo1, bar: bar1, baz: baz1 }
+                   ^ Avoid the use of double negation (`!!`).
+            do_something
+          else
+            { foo: !!foo2, bar: bar2, baz: baz2 }
+                   ^ Avoid the use of double negation (`!!`).
+            do_something
+          end
+        end
+      RUBY
+
+      expect_correction(<<~RUBY)
+        def foo?
+          if condition_foo?
+            { foo: !foo0.nil?, bar: bar0, baz: baz0 }
+            do_something
+          elsif condition_bar?
+            { foo: !foo1.nil?, bar: bar1, baz: baz1 }
+            do_something
+          else
+            { foo: !foo2.nil?, bar: bar2, baz: baz2 }
+            do_something
+          end
+        end
+      RUBY
+    end
+
+    it 'registers an offense and corrects for `!!` with array when not return location' \
+       'and using `if`, `elsif`, and `else`' do
+      expect_offense(<<~RUBY)
+        def foo?
+          if condition_foo?
+            [ !!foo0, bar0, baz0 ]
+              ^ Avoid the use of double negation (`!!`).
+            do_something
+          elsif condition_bar?
+            [ !!foo1, bar1, baz1 ]
+              ^ Avoid the use of double negation (`!!`).
+            do_something
+          else
+            [ !!foo2, bar2, baz2 ]
+              ^ Avoid the use of double negation (`!!`).
+            do_something
+          end
```

Comment:
Can you remove spaces inside the array literal brace?

```suggestion
          if condition_foo?
            [!!foo0, bar0, baz0]
              ^ Avoid the use of double negation (`!!`).
            do_something
          elsif condition_bar?
            [!!foo1, bar1, baz1]
              ^ Avoid the use of double negation (`!!`).
            do_something
          else
            [!!foo2, bar2, baz2]
              ^ Avoid the use of double negation (`!!`).
            do_something
          end
```